src/vueWrapper.ts

```diff
@@ -38,7 +50,124 @@ export class VueWrapper<T extends ComponentPublicInstance> {
     return this.vm.$el.parentElement
   }
 
+  /**
+   * Get all the target elements defined in the teleportTarget mount option.
+   *
+   * @private
+   *
+   * @return {Element[]|undefined}
+   */
+  private getTeleportElements(): Element[] | undefined | never {
+    if (!this.teleportTargets.length) {
+      return
+    }
+
+    const elements: Element[] = []
+
+    this.teleportTargets.forEach((target) => {
+      if (typeof target !== 'string') {
+        elements.push(target)
+        return
+      }
+
+      const element = document.querySelector(target)
+
+      if (!element) {
+        throw new Error("The teleport target '" + target + "' cannot be found.")
+      }
+
+      elements.push(element)
+    })
+
+    return elements
+  }
+
+  /**
+   * Get the teleport target element for the given vue instance.
+   *
+   * @param {ComponentPublicInstance} vm
+   *
+   * @private
+   *
+   * @return {Element}
+   */
+  private getTeleportElement(
+    vm: ComponentPublicInstance = this.componentVM
+  ): Element | never {
+    const elements = this.getTeleportElements()
+    const errorMsg =
+      vm.$.type.name +
+      "'s teleport target element cannot be found. Are you sure the target exits in the DOM?"
+
+    if (!elements || !elements.length) {
+      throw new Error(errorMsg)
+    }
+
+    const index = elements.findIndex((element) => {
+      // todo what if the component has nor been realised as it's hidden by <!--v-if-->
+      // get the element of which the given vm is its direct child
+      return Array.from(element.childNodes).some(
+        (
+          child: Element & { __vueParentComponent: ComponentInternalInstance }
+        ) => {
+          return (
+            child.hasOwnProperty('__vueParentComponent') &&
+            child.__vueParentComponent.uid === vm.$.uid
+          )
+        }
+      )
+    })
+
+    if (index === -1) {
+      throw new Error(errorMsg)
```

Comment:
Shouldn't that be a different error message? I think it would be less confusing to distinguish the two cases.